### PR TITLE
swarm: README.md

### PR DESCRIPTION
**Pull request by nonsense**
I'd like to merge this README.md under the `swarm/` folder with the purpose of it being the main entrypoint for Swarm, alongside our website (https://swarm.ethereum.org). I think it would be nice to have a README.md close to the code, where we can add useful commands, links to documentation, and basically a general "Getting Started" guide. I think the main documentation - https://swarm-guide.readthedocs.io - should be a comprehensive guide, whereas this is just a quick entrypoint for new people.

I think this should also make it clear that `ethereum/go-ethereum` is the source of truth for the codebase, and that `ethersphere` is a fork for the core team to communicate and organise better.

### CODE REVIEW

File: swarm/README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+procedures quick and simple.
+
+Please make sure your contributions adhere to our coding guidelines:
+
+ * Code must adhere to the official Go [formatting](https://golang.org/doc/effective_go.html#formatting) guidelines (i.e. uses [gofmt](https://golang.org/cmd/gofmt/)).
+ * Code must be documented adhering to the official Go [commentary](https://golang.org/doc/effective_go.html#commentary) guidelines.
+ * Pull requests need to be based on and opened against the `master` branch.
+ * Commit messages should be prefixed with the package(s) they modify.
+   * E.g. "swarm/fuse: ignore default manifest entry"
+
```
Comment (janos) [inline swarm/README.md]: Maybe to mention https://github.com/ethereum/go-ethereum/wiki/Code-Review-Guidelines, also?

**Review by janos — APPROVED** [review]
LGTM

File: swarm/README.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+early feedback which can make both your efforts much lighter as well as our review and merge
+procedures quick and simple.
+
+Please make sure your contributions adhere to our coding guidelines:
+
+ * Code must adhere to the official Go [formatting](https://golang.org/doc/effective_go.html#formatting) guidelines (i.e. uses [gofmt](https://golang.org/cmd/gofmt/)).
+ * Code must be documented adhering to the official Go [commentary](https://golang.org/doc/effective_go.html#commentary) guidelines.
+ * Pull requests need to be based on and opened against the `master` branch.
+ * [Code review guidelines](https://github.com/ethereum/go-ethereum/wiki/Code-Review-Guidelines).
+ * Commit messages should be prefixed with the package(s) they modify.
```
Comment (zelig) [inline swarm/README.md:43]: and must not exceed 72 characters

File: swarm/README.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+
+## Contribution
+
+Thank you for considering to help out with the source code! We welcome contributions from
+anyone on the internet, and are grateful for even the smallest of fixes!
+
+If you'd like to contribute to Swarm, please fork, fix, commit and send a pull request
+for the maintainers to review and merge into the main code base. If you wish to submit more
+complex changes though, please check up with the core devs first on [our Swarm gitter channel](https://gitter.im/ethereum/swarm)
```
Comment (zelig) [inline swarm/README.md]: i would put orange lounge

File: swarm/README.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
+## Swarm
+
+[https://swarm.ethereum.org](https://swarm.ethereum.org)
+
+Swarm is a distributed storage platform and content distribution service, a native base layer service of the ethereum web3 stack. The primary objective of Swarm is to provide a decentralized and redundant store for dapp code and data as well as block chain and state data. Swarm is also set out to provide various base layer services for web3, including node-to-node messaging, media streaming, decentralised database services and scalable state-channel infrastructure for decentralised service economies.
+
+[![Travis](https://travis-ci.org/ethereum/go-ethereum.svg?branch=master)](https://travis-ci.org/ethereum/go-ethereum)
+[![Gitter](https://badges.gitter.im/Join%20Chat.svg)](https://gitter.im/ethereum/swarm?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge)
```
Comment (zelig) [inline swarm/README.md]: orange lounge too

File: swarm/README.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+If you'd like to contribute to Swarm, please fork, fix, commit and send a pull request
+for the maintainers to review and merge into the main code base. If you wish to submit more
+complex changes though, please check up with the core devs first on [our Swarm gitter channel](https://gitter.im/ethereum/swarm)
+to ensure those changes are in line with the general philosophy of the project and/or get some
+early feedback which can make both your efforts much lighter as well as our review and merge
+procedures quick and simple.
+
+Please make sure your contributions adhere to our coding guidelines:
+
+ * Code must adhere to the official Go [formatting](https://golang.org/doc/effective_go.html#formatting) guidelines (i.e. uses [gofmt](https://golang.org/cmd/gofmt/)).
```
Comment (zelig) [inline swarm/README.md:39]: there could be a lot more info here see swarm onboarding doc, dev guide, contribution guidelines

**Review by zelig — COMMENTED** [review]
great endeavor to put this in the code but i feel this is too lean.
especially that it does not even link    to 

* developers guide
* onboarding doc
* contribution guidelines
* dev relevant sections of the swarm guide

File: swarm/README.md (hunk 3 above)
Comment (nonsense) [reply swarm/README.md]: Done.

File: swarm/README.md (hunk 5 above)
Comment (nonsense) [reply swarm/README.md:39]: Sure, we could enhance it further. I just wanted to have the minimum necessary as a start.

1. Contribution guidelines are here.
2. Onboarding document - I don't think it belongs on the front page of the README. We should also update that document. I checked it today and it is a bit outdated.
3. Dev guide - I am not sure what you mean by that?

File: swarm/README.md (hunk 4 above)
Comment (nonsense) [reply swarm/README.md]: Done.

File: swarm/README.md (hunk 2 above)
Comment (nonsense) [reply swarm/README.md:43]: Actually it is 80 characters. I think by default it is set to 72 characters in Gitcop (only on ethersphere side) and Balint has the credentials to change that. Basically ethersphere Gitcop is misconfigured at the moment, and not the same as ethereum repo.

I will actually ask Peter, why we have this in the first place, when we squash commits on merge... this limit is a bit redundant.

File: swarm/README.md (hunk 5 above)
Comment (zelig) [reply swarm/README.md:39]: how about linking to and consolidate the following docs? 

* https://github.com/ethersphere/go-ethereum/wiki/Git-Workflow
* https://github.com/ethersphere/swarm/wiki/swarm
* https://github.com/ethereum/go-ethereum/wiki/Developers'-Guide 
* https://github.com/ethereum/go-ethereum#contribution

Comment (nonsense) [reply swarm/README.md:39]: I think we can do this step by step, however:
https://github.com/ethersphere/go-ethereum/wiki/Git-Workflow - I don't think we need this anymore
https://github.com/ethersphere/swarm/wiki/swarm - the URL and the whole document has to be refactored, but I agree we need an Onboarding document.
https://github.com/ethereum/go-ethereum/wiki/Developers'-Guide - I have to think about this one. We probably need something like this for Swarm.
https://github.com/ethereum/go-ethereum#contribution - this is already part of our README.

Bottom line - I suggest we merge as is, and continue with fixing the Developers' Guide and the Onboarding document. Actually they could be one and the same document, as they overlap a lot. Both in Onboarding and in Developers' Guide, you have to explain:
- how to setup dev environment
- how to use some of the tools that we use that are mostly hidden (logging, clusters, metrics, etc.)
- how to test in Go (tests vs benchmarks)
- how to use `janos/swarm-test-cluster`
- etc.

**Comment by nonsense** [comment]
closing this here, as it is open and will be merged to ethereum/go-ethereum. next step is to do refactor of our wiki pages and structure them a bit better.
